pkg/bitwarden/cli_test.go

```diff
@@ -0,0 +1,561 @@
+}`), fmt.Errorf("failed to login: Username or password is incorrect. Try again")
+			},
+			expectedError: fmt.Errorf("failed to login: Username or password is incorrect. Try again"),
+		},
+		{
+			name:     "already logged in",
+			username: "u",
+			password: "p",
+			run: func(cmd string, args ...string) ([]byte, error) {
+				if cmd != "bw" || !reflect.DeepEqual(args, []string{"login", "u", "p", "--response"}) {
+					return nil, fmt.Errorf("unexpected call")
+				}
+				return []byte(`{
+  "success": false,
+  "message": "You are already logged in as dptp@redhat.com."
+}`), fmt.Errorf("failed to login: You are already logged in as dptp@redhat.com")
+			},
+			expectedError: fmt.Errorf("failed to login: You are already logged in as dptp@redhat.com"),
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			actual, actualError := newCliClientWithRun(tc.username, tc.password, tc.run)
+			equalError(t, tc.expectedError, actualError)
+			actualClient, ok := actual.(*cliClient)
+			if !ok {
+				t.Errorf("Failed to cast to bitwardenCliClient")
+			}
+			equal(t, tc.expectedSession, actualClient.session)
+			equal(t, tc.expectedSavedItems, actualClient.savedItems)
+		})
+	}
+}
+
+func TestLogout(t *testing.T) {
+	testCases := []struct {
+		name     string
+		bwClient Client
+		expected error
+	}{
+		{
+			name: "basic case",
+			bwClient: &cliClient{
+				username: "u",
+				password: "p",
+				run: func(cmd string, args ...string) ([]byte, error) {
+					if cmd != "bw" || !reflect.DeepEqual(args, []string{"logout", "--response"}) {
+						return nil, fmt.Errorf("unexpected call")
+					}
+					return []byte(`{
+  "success": true,
+  "data": {
+    "noColor": false,
+    "object": "message",
+    "title": "You have logged out.",
+    "message": null
+  }
+}`), nil
+				},
+			},
+		},
+		{
+			name: "command not find",
+			bwClient: &cliClient{
+				run: func(cmd string, args ...string) ([]byte, error) {
+					if cmd != "bw" || !reflect.DeepEqual(args, []string{"logout", "--response"}) {
+						return nil, fmt.Errorf("unexpected call")
+					}
+					return []byte(""), fmt.Errorf("command not found in $PATH")
+				},
+			},
+			expected: fmt.Errorf("command not found in $PATH"),
+		},
+		{
+			name: "not logged in",
+			bwClient: &cliClient{
+				run: func(cmd string, args ...string) ([]byte, error) {
+					return []byte(`{
+  "success": false,
+  "message": "You are not logged in."
+}`), fmt.Errorf("failed to login: You are not logged in")
+				},
+			},
+			expected: fmt.Errorf("failed to login: You are not logged in"),
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			actual := tc.bwClient.Logout()
+			equalError(t, tc.expected, actual)
+		})
+	}
+}
+
+func TestGetFieldOnItem(t *testing.T) {
+	testCases := []struct {
+		name        string
+		client      *cliClient
+		itemName    string
+		fieldName   string
+		expected    string
+		expectedErr error
+	}{
+		{
```

Comment:
Think about your formatting. This test case is large and spans 50 lines., which makes it hard to read. Try something like this:


```go
		{
			name: "basic case",
			client: &cliClient{savedItems: []Item{{
						ID:   "id1", Name: "unsplash.com",
						Fields: []Field{{Name:  "API Key", Value: "value1"},{Name:  "no name",Value: "value2"}},
				},},
			},
			itemName:  "unsplash.com",
			fieldName: "API Key",
			expected:  "value1",
		},
```

Furthermore, you can just have the list of items in your test case and create the client from that list at runtime, further making this test case simpler.

Reply:
Fix the client for all test cases.